bundles/org.eclipse.mita.program/src/org/eclipse/mita/program/inferrer/ElementSizeInferrer.xtend

```diff
@@ -137,12 +137,9 @@ class ElementSizeInferrer {
 	}
 	
 	protected def dispatch ElementSizeInferenceResult doInfer(ElementReferenceExpression obj) {
-		val objType = typeInferrer.infer(obj);
-		if(objType?.type instanceof GeneratedType) {
-			val inferredSize = obj.inferFromType;
-			if(inferredSize instanceof ValidElementSizeInferenceResult) {
-				return inferredSize;
-			}
+		val inferredSize = obj.inferFromType;
+		if (inferredSize instanceof ValidElementSizeInferenceResult) {
+			return inferredSize;
```

Comment:
@wegendt-bosch I am not sure about this one. But why do we need to check for GeneratedType here when it is also handled in inferFromType method. Now the logic is simply: Try to infer based on reference expression, if not possible, infer based on the referenced element itself.

Reply:
Looking at the history of this code, it seems that the whole if was made to cater to function returning generated types. To not change behaviour too much this was only applied to exactly those. If it works fine this way, too, all the better.